tools/Microsoft.AspNetCore.Server.Kestrel.GeneratedCode/KnownHeaders.cs

```diff
@@ -547,14 +639,16 @@ public bool MoveNext()
             {{
                 switch (_state)
                 {{
-                    {Each(loop.Headers, header => $@"
-                        case {header.Index}:
-                            goto state{header.Index};
+                    {Each(loop.Headers.Where(header => header.Index >= 0), header => $@"
```

Comment:
Using `Index = -1` to distinguish the Content-Length header seems a little weird. Couldn't you just filter on `Name`?